glib/src/source.rs

```diff
@@ -142,6 +142,56 @@ fn into_raw_unix_fd<F: FnMut(RawFd, IOCondition) -> Continue + 'static>(func: F)
     Box::into_raw(func) as gpointer
 }
 
+///
+/// In comparisong to `timeout_add()`, this only requires `func` to be
+/// `FnOnce`, and will automatically return `Continue(false)`.
+where
+    F: FnOnce() + Send + 'static,
```

Comment:
Something seemed to have gone missing here

Reply:
No, that's my copy-paste blueprint that I accidentally commited as well.